### PR TITLE
Add scripts to build statically-linked htslib

### PR DESCRIPTION
**Pull request by jdblischak**
These scripts will build a statically-linked htslib under msys2 for version 1.15.1, create a GitHub release, and upload the release tarball as an asset. Thanks to @dhoke4tdb for the patches

### CODE REVIEW

File: .github/workflows/msys2-htslib-build.yml
Comment (dhoke4tdb) [inline .github/workflows/msys2-htslib-build.yml:15]: Unsure if this is separate from PACKAGE_VERSION elsewhere (multiple namespaces?), but I observe that here it is 1.15.1 but in https://github.com/jdblischak/m2w64-htslib-build/blob/build/scripts/download-htslib.sh it seems to be, (possibly prefixed by something else), 1.16....
1.15.1 shows up as well in https://github.com/jdblischak/m2w64-htslib-build/blob/build/scripts/bundle-htslib.sh
1.15.1 again in https://github.com/jdblischak/m2w64-htslib-build/blob/build/scripts/release.sh

**Review by ihnorton — APPROVED** [review]
cc @dhoke4tdb for a second look.

File: README.md
Comment (dhoke4tdb) [inline README.md:4]: Is it worth defining what 'statically linked' means in this situation, since it's not entirely statically linked...

If decide a good idea to define, here's my first attempt...
*'statically linked' here implies static linkage to all needed msys2 libraries that are not referencing .DLLs that are provided by windows.

File: scripts/build-htslib.sh
Comment (dhoke4tdb) [inline scripts/build-htslib.sh:11]: There are a couple of additional source patches (for .h) files that should probably be applied...
Note that two apply to 1.15.1, and only one applies to 1.16 (as the other change already appears in 1.16), not sure whether they were previously provided to you or not.
files are named...
htslib.1.15.1.hts_defs.h.patch
htslib.1.15.1.vcf.h.patch

and htslib.1.15.1.vcf.h.patch can also be applied to 1.16

**Review by dhoke4tdb — CHANGES_REQUESTED** [review]
Uncertainty over actual htslib version retrieved versus version numbers used everywhere else, maybe this is not a problem, but the 1.16 is mentally bothersome amidst all the other 1.15.1's.

The missing patches are against the headers which are installed from the assembled htslib package, so could be done later, but maybe better to do here just to keep the patching in one place.

File: .github/workflows/msys2-htslib-build.yml
Comment (jdblischak) [reply .github/workflows/msys2-htslib-build.yml:15]: This `PACKAGE_VERSION` env var takes precedence. The syntax `${PACKAGE_VERSION-1.16}` means to fall back to "1.16" only if the env var `PACKAGE_VERSION` is empty (which is never the case in the GitHub Actions workflow). However, I will change it to 1.15.1 as well for consistency

File: README.md
Comment (jdblischak) [reply README.md:4]: Good point, I'll add your expanded definition to the README

File: scripts/build-htslib.sh
Comment (jdblischak) [reply scripts/build-htslib.sh:11]: > There are a couple of additional source patches (for .h) files that should probably be applied...

I wasn't explicitly sent these additional patches. You have multiple branches in TileDB-VCF. Could you please send me links to the exact patch files I should apply?

> and htslib.1.15.1.vcf.h.patch can also be applied to 1.16

Also, just to confirm, my understanding is that the plan is to first build htslib 1.15.1 for use with tiledb-vcf-feedstock, and then later update to 1.16

**Comment by jdblischak** [comment]
@ihnorton This PR is blocking the next step of creating the feedstock to generate conda binaries. Could you please weigh in on the following?

* Should I wait for David's additional patches before merging, or merge and add the patches later?
* Are we targeting a build of htslib 1.15.1 first or should we skip straight to 1.16?

**Comment by ihnorton** [comment]
I'm fine with merging this.

> Are we targeting a build of htslib 1.15.1 first

Yes, we should target the same version as TileDB-VCF for now.

**Comment by dhoke4tdb** [comment]
bumping patches, they need to be incorporated into the source for the msys2
htslib build.

**Comment by jdblischak** [comment]
Pending in PR #2

**Pull request merged**
